[cxx-interop] Add basic C++ stdlib overlay

egorzhdan opened the pull request:
This will allow us to add Swift conformances to C++ standard library types, and improve usability of the C++ standard library from Swift.

egorzhdan committed:
[cxx-interop] Add basic C++ stdlib overlay

This will allow us to add Swift conformances to C++ standard library types, and improve usability of the C++ standard library from Swift.

zoecarver commented on test/Interop/Cxx/stdlib/overlay.swift:
```diff
@@ -0,0 +1,21 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-experimental-cxx-interop)
```
Nit: maybe it makes more sense to have this be `Cxx/stdlib/overlay/string.swift` or `Cxx/overlay/string.swift`. WDYT?

egorzhdan replied:
Yeap, let's make it `Cxx/stdlib/overlay/string.swift`

zoecarver reviewed (approved):
Super exciting!

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please smoke test macOS

egorzhdan commented:
@swift-ci please smoke test Linux

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please smoke test

egorzhdan committed:
[cxx-interop] Emit `-enable-experimental-cxx-interop` into the serialized module interface

`.swiftinterface` files contain a list of compiler flags that were used to build the module. For example, the list includes `-enable-objc-interop`. It should also include `-enable-experimental-cxx-interop`, since it affects the way the module should be imported.

egorzhdan commented:
I added one more commit with a general C++ interop change, not specific to the overlay, to fix `verify_all_overlays.py`.
@zoecarver can you please take another look?

egorzhdan commented:
Discussed offline last week, will merge this as soon as the tests pass (re-rerunning to make sure libswift doesn't break).

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please test Windows

egorzhdan commented:
@swift-ci please smoke test

egorzhdan commented:
@swift-ci please smoke test
